### PR TITLE
Replace deprecated python settings with extensions (GSI-411)

### PR DESCRIPTION
**Pull request by TheByronHimes**
According to [this post](https://github.com/microsoft/vscode-python/wiki/Migration-to-Python-Tools-Extensions), there are several settings that are soon to be deprecated for the python configuration in VS Code. Most of them are around linting, the effect being that we need to use comparable extensions instead of the configured vs-code python devcontainer add-ins. 
We needed to remove some of these anyway, such as bandit and isort. 

> Note: I also removed the Postgres settings since we don't use that.

The added extensions are:
- [ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)
  - Will now highlight issues in the editor and offer quickfixes or ignores
- [black](https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter)
  - Configured to run on save like before
- [mypy](https://marketplace.visualstudio.com/items?itemName=ms-python.mypy-type-checker)

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -73,7 +50,10 @@
 				"yzhang.markdown-all-in-one",
 				"visualstudioexptteam.vscodeintellicode",
 				"ymotongpoo.licenser",
-				"editorconfig.editorconfig"
+				"editorconfig.editorconfig",
```
Comment (KerstenBreuer) [inline .devcontainer/devcontainer.json:53]: Are we still using the editor config? Or can we move that out? I.e. also the .editorconfig in the root directory. As far as I understand this is only relevant if we are using multiple editors which we are not.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -60,7 +38,6 @@
 				"mikestead.dotenv",
 				"ms-azuretools.vscode-docker",
 				"ms-python.python",
-				"ms-python.isort",
```
Comment (KerstenBreuer) [inline .devcontainer/devcontainer.json:63]: Have you also checked if it is still best practice to use "ms-python.python" along with "ms-python.vscode-pylance" for general python language support?

**Review by KerstenBreuer — COMMENTED** [review]
Thanks a lot. Looks good. Just having two questions.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (TheByronHimes) [reply .devcontainer/devcontainer.json:63]: Ah, no it looks like pylance gets automatically installed with the ms python extension now, so no need to independently specify it.

Comment (KerstenBreuer) [reply .devcontainer/devcontainer.json:63]: OK, perfect.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (KerstenBreuer) [reply .devcontainer/devcontainer.json:53]: So for end of line settings, you can add this to vscode settings:
```
    "files.eol": "\n"
```
When it comes to the encoding, the default is already utf-8 in vscode. So no need to set that.

So I guess we can remove the editorconfig

**Review by mephenor — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (TheByronHimes) [reply .devcontainer/devcontainer.json]: @KerstenBreuer please see [0854d59](https://github.com/ghga-de/microservice-repository-template/pull/171/commits/0854d59fb63291ca385c024c590bf55acf584cf6)

Comment (KerstenBreuer) [reply .devcontainer/devcontainer.json]: Perfect, thanks

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**